skbio/metadata/tests/test_intersection.py

```diff
@@ -0,0 +1,222 @@
+        self.intervals = iv
+
+    def test_upstream(self):
+        iv = self.intervals
+        upstreams = iv.upstream_of_interval(IntervalObj(59, 60),
+                                            num_intervals=200)
+
+        for u in upstreams:
+            self.assertTrue(u.end < 59)
+
+        upstreams = iv.upstream_of_interval(IntervalObj(60, 70, strand=-1),
+                                            num_intervals=200)
+        for u in upstreams:
+            self.assertTrue(u.start > 70)
+
+        upstreams = iv.upstream_of_interval(IntervalObj(58, 58, strand=-1),
+                                            num_intervals=200)
+        for u in upstreams:
+            self.assertTrue(u.start > 59)
+
+    def test_downstream(self):
+        iv = self.intervals
+        downstreams = iv.downstream_of_interval(IntervalObj(59, 60),
+                                                num_intervals=200)
+        for d in downstreams:
+            self.assertTrue(d.start > 60)
+
+        downstreams = iv.downstream_of_interval(IntervalObj(59, 60, strand=-1),
+                                                num_intervals=200)
+        for d in downstreams:
+            self.assertTrue(d.start < 59)
+
+    def test_n(self):
+        iv = self.intervals
+        for i in range(0, 90, 10):
+            r = iv.after(i, max_dist=20, num_intervals=2)
+            self.assertEqual(r[0].start, i + 10)
+            self.assertEqual(r[1].start, i + 20)
+
+            r = iv.after_interval(IntervalObj(i, i),
+                                  max_dist=20, num_intervals=2)
+            self.assertEqual(r[0].start, i + 10)
+            self.assertEqual(r[1].start, i + 20)
+
+
+class LotsaTestCase(unittest.TestCase):
+    """ put lotsa data in the tree and make sure it works"""
+    def setUp(self):
+        iv = IntervalNode(1, 2, IntervalObj(1, 2))
+        self.max = 1000000
+        for i in range(0, self.max, 10):
+            f = IntervalObj(i, i)
+            iv = iv.insert(f.start, f.end, f)
+
+        for i in range(600):
+            iv = iv.insert(0, 1, IntervalObj(0, 1))
+        self.intervals = iv
+
+    def test_count(self):
+        iv = self.intervals
+
+        r = iv.right(1, n=33)
+        self.assertEqual(len(r), 33)
+
+        l = iv.left(1, n=33)
+        self.assertEqual(len(l), 1)
+
+        u = iv.right(1, n=9999)
+        self.assertEqual(len(u), 250)
+
+        # now increase max_dist
+        u = iv.right(1, n=9999, max_dist=99999)
+        self.assertEqual(len(u), 9999)
+
+    def test_max_dist(self):
+        iv = self.intervals
+        r = iv.right(1, max_dist=0, n=10)
+        self.assertEqual(len(r), 0)
+
+        for n, d in enumerate(range(10, 1000, 10)):
+            r = iv.right(1, max_dist=d, n=10000)
+            self.assertEqual(len(r), n + 1)
+
+    def test_find(self):
+        iv = self.intervals
+        path = sys.path[:]
+        sys.path = sys.path[2:]
+        random = __import__("random")
```

Comment:
You can just use `import random` at the top of this test (PEP8 is fine with it as long as it comes before code in the function). If you do ever find yourself needing to dynamically import, use `importlib.import_module`, it's a more supported API.